### ISSUE TITLE
Calling set_process(false) when NOTIFICATION_EXIT_TREE emits error in terminal

### ISSUE THREAD

**Issue by dhoverml** [godotengine/godot#81159]
### Godot version

v4.1.1.stable.official [bd6af8e0e]

### System information

Ubuntu 20.04 LTS

### Issue description

See title.

### Steps to reproduce

1. Create a new scene with one node
2. Attach a script to any node type that includes the following:
```
func _process(delta):
  pass


func _notification(what):
  if what == NOTIFICATION_EXIT_TREE:
    set_process(false)
```
4. Run this command in the terminal (replace `./path/to/godot` and `path/to/project` with valid paths on your machine)
```
./path/to/godot --headless --quit --path path/to/project
```
5. See this error in the terminal
```
ERROR: Condition "!found" is true.
   at: _remove_node_from_process_group (scene/main/scene_tree.cpp:1122)
```

**Notes**
1. I understand that it is redundant to call `set_process(false)` when exiting the tree, since `_process(delta)` won't be called until it is added back.  However I think this is still a bug since it's valid code.
2. The error occurs every time the node is removed from the scene tree (assuming it calls `set_process(true)` sometime before it is removed again), not just during shutdown.
3. The error only appears when run via the command line.  It does not appear in the editor's console.
4. Running this in the editor (with `--editor`) will also emit the error in the terminal and still not in the editor's console.
5. You don't have to run with `--headless --quit`.  Removing the two, then manually closing the window, will also repro the bug.

### Minimal reproduction project

[set_process_bug.zip](https://github.com/godotengine/godot/files/12475459/set_process_bug.zip)

**Comment by YuriSizov** [comment]
> I understand that it is redundant to call `set_process(false)` when exiting the tree, since `_process(delta)` won't be called until it is added back.  However I think this is still a bug since it's valid code.

The call is indeed redundant, and the logged error is a sign to the developer that it's redundant. The fact that an error prints is not a bug, it means that the case is correctly handled to avoid errors, but the caller does something wrong. We could try to improve the message, of course.

**Comment by dhoverml** [comment]
Agreed.  Printing the error in the editor's console would help I think.
